behandlingsprosess/src/test/java/no/nav/ung/sak/domene/behandling/steg/beregning/UngdomsytelseBeregnDagsatsTest.java

```diff
@@ -103,106 +103,159 @@ void setUp() {
-        assertThat(first.getValue().dagsats()).isEqualByComparingTo("608.31");
+        assertThat(first.getValue().dagsats()).isEqualByComparingTo(BigDecimal.valueOf(608.31));
 
         var second = iterator.next();
-        assertThat(second.getFom()).isEqualTo(førsteDagMedHøySats);
-        assertThat(second.getTom()).isEqualTo(tom);
+        assertThat(second.getFom()).isEqualTo(førsteDagMedHøySats.plusDays(1));
+        assertThat(second.getTom()).isEqualTo(tom.minusMonths(1));
         assertThat(second.getValue().grunnbeløpFaktor()).isEqualByComparingTo(BigDecimal.valueOf(2));
-        assertThat(second.getValue().grunnbeløp()).isEqualByComparingTo(BigDecimal.valueOf(124028));
-        assertThat(second.getValue().dagsats()).isEqualByComparingTo(BigDecimal.valueOf(954.06));
+        assertThat(second.getValue().grunnbeløp()).isEqualByComparingTo(BigDecimal.valueOf(118620));
+        assertThat(second.getValue().dagsats()).isEqualByComparingTo(BigDecimal.valueOf(912.46));
+
+        // Etter G-regulering
+        var third = iterator.next();
+        assertThat(third.getFom()).isEqualTo(datoForGRegulering);
+        assertThat(third.getTom()).isEqualTo(tom);
+        assertThat(third.getValue().grunnbeløpFaktor()).isEqualByComparingTo(BigDecimal.valueOf(2));
+        assertThat(third.getValue().grunnbeløp()).isEqualByComparingTo(BigDecimal.valueOf(124028));
+        assertThat(third.getValue().dagsats()).isEqualByComparingTo(BigDecimal.valueOf(954.06));
     }
 
 
     @Test
     void skal_beregne_dagsats_for_en_periode_med_start_i_mars_2024_og_slutt_i_mai_2024_og_bruker_blir_25_år_første_april() {
         var fom = LocalDate.of(2024, 3, 1);
         var tom = LocalDate.of(2024, 5, 30);
+        var datoForGRegulering = LocalDate.of(2024, 5, 1);
         var perioder = new LocalDateTimeline<>(fom, tom, Boolean.TRUE);
         var tjuefemårsdag = LocalDate.of(2024, 4, 1);
         var fødselsdato = tjuefemårsdag.minusYears(25);
         var dagsatsTidslinje = tjeneste.beregnDagsats(null, perioder, fødselsdato, LocalDate.now(), true);
 
+        var segmenter = dagsatsTidslinje.resultatTidslinje().toSegments();
+        assertThat(segmenter.size()).isEqualTo(3);
+
+        var iterator = segmenter.iterator();
+        var first = iterator.next();
+        assertThat(first.getFom()).isEqualTo(fom);
+        var sisteDagMedLavSats = tjuefemårsdag.minusDays(1);
+        assertThat(first.getTom()).isEqualTo(sisteDagMedLavSats);
+        assertThat(first.getValue().grunnbeløpFaktor()).isEqualByComparingTo(BigDecimal.valueOf(1.33333));
+        assertThat(first.getValue().grunnbeløp()).isEqualByComparingTo(BigDecimal.valueOf(118620));
+        assertThat(first.getValue().dagsats()).isEqualByComparingTo(BigDecimal.valueOf(608.31));
+
+        var second = iterator.next();
+        assertThat(second.getFom()).isEqualTo(datoForGRegulering.minusMonths(1));
```

Comment:
Det gir vel kanskje mest meining å bruke tjuefemårsdag i staden for datoForGRegulering.minusMonths(1), sjølv om dei tilfeldigvis er like for akkurat denne testen.